### DISCUSSION TITLE
Such as "System.Console" and  "Internal.Console". How do I filter these internal namespaces when reflecting?

### DISCUSSION THREAD

**Discussion by NMSAzulX** [dotnet/runtime#85503] [Q&A]
I am reflecting all the namespaces in Assembly. But I got `Internal`, which conflicts with `System`.
![image](https://user-images.githubusercontent.com/16236903/235046524-59164000-da97-4e2c-9edd-8ce1f4d0ad01.png)
It has no any flag when reflecting. How do I filter these internal namespaces?

```c#
var types = assembly.ExportedTypes;
 if (types == null)
 {
      return;
}
foreach (var type in types)
{

      var name = type.Namespace;
      if (name == null)
      {
            continue;
      }
      var a = assembly.GetType(type.FullName);
      if (name.StartsWith("Internal"))
      {
      }

}
```

**Comment by huoyaoyuan** [comment]
What's your purpose of the reflection?
Internal members in BCL will change and have changed release by release. Whenever you want to build a strong function that works for different versions, you should filter by their visibility.

**Reply by NMSAzulX** [reply]
Do sth. How do I filter these internal namespaces?

**Reply by NMSAzulX** [reply]
Dynamic Compile

**Reply by huoyaoyuan** [reply]
Namespace is not a thing after compiling to IL. You can only filter them with the prefix of full name.

**Reply by NMSAzulX** [reply]
这里应该是 SDK 引用程序集和实际程序集的差异, 引用程序集不包含这些, 所以平时开发没问题,  实际程序集包含的东西太多, 也是比较真实, 如果我都从真实程序集中获取程序集而不是加载和携带引用程序集, 动态编译的内存占用问题可以得到改善.
